hack/kind-cluster-build.sh

```diff
@@ -150,6 +159,9 @@ ${KIND_BIN} create cluster --config ${configFile} --image kindest/node:${k8sVers
 ${KIND_BIN} get kubeconfig --name=${clusterName} > ${kubeconfigPath}
 export KUBECONFIG=${kubeconfigPath}
 
+echo "connect the local docker registry to the cluster network"
+docker network connect "kind" "${reg_name}"
```

Comment:
Delete kind cluster would not remove the connection, and restart kind would comes with the error:
```
Error response from daemon: endpoint with name registry already exists in network kind
``` 

I think we need to see if it's already connected before this step, or check the error output

Reply:
@Yiyiyimu thanks for spotting this! I updated my PR to check the err and exit code.